fanctrl.py

```diff
@@ -429,9 +434,31 @@ def isOnAC(self):
 
 
 class EctoolHardwareController(HardwareController, ABC):
+    noBatteryMode = False
+    nonBatterySensors = None
+    
+    def __init__(self, noBatteryMode=False):
+        self.noBatteryMode = noBatteryMode
+        if self.noBatteryMode:
+            self.getNonBatterySensors()
+    
+    def getNonBatterySensors(self):
```

Comment:
When a function is named "getxxxx", I expect a value to be returned and that no global (or class in this case) variable assignment is made.
Perhaps we should instead assign the "nonBatterySensors" in the init or in an other accordingly named method and return the filtered values?

Reply:
i don't much like having so much much code in the init function, so would the function name `populateNonBatterySensors` be alright for this?